### PR TITLE
[limonp]  Add new port

### PR DESCRIPTION
**Pull request by playgithub**
**Describe the pull request**

- #### What does your PR fix?
  New Port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
  All

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by dg0yt** [comment]
- Given the minimal changes wrt to the source of your fork, you can simply use patches and/or a custom `CMakeLists.txt`. It is header-only, and vcpkg doesn't want to build tests.
- The CMake config export is not from the source of your fork, so it should be moved to namespace "unofficial".
- I fail to see how the added license and copyright are authorized by the source of your fork.
- Please don't close PRs to open new ones for the same topic. Just force-push your updated branch.

File: ports/limonp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO playgithub/limonp
```
Comment (FrankXie05) [inline ports/limonp/portfile.cmake:3]:
```suggestion
    REPO yanyiwu/limonp

```
This is not an official repo

File: ports/limonp/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
+
+# # Moves all .cmake files from /debug/share/limonp/ to /share/limonp/
+# # See /docs/maintainers/ports/vcpkg-cmake-config/vcpkg_cmake_config_fixup.md for more details
+# When you uncomment "vcpkg_cmake_config_fixup()", you need to add the following to "dependencies" vcpkg.json:
+#{
+#    "name": "vcpkg-cmake-config",
+#    "host": true
+#}
+# vcpkg_cmake_config_fixup(CONFIG_PATH cmake TARGET_PATH share/limonp)
```
Comment (FrankXie05) [inline ports/limonp/portfile.cmake:23]: No need for this annotation
```suggestion
```

File: ports/limonp/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (FrankXie05) [inline ports/limonp/portfile.cmake:25]:
```suggestion
file(INSTALL "${SOURCE_PATH}/README.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

```
No upstream licenses need to be installed

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/limonp/portfile.cmake (hunk 1 above)
Comment (playgithub) [reply ports/limonp/portfile.cmake:3]: Yes, `CMakeListst.txt` refactored

File: ports/limonp/portfile.cmake (hunk 3 above)
Comment (playgithub) [reply ports/limonp/portfile.cmake:17]: So a fake one?

**Comment by playgithub** [comment]
> * Given the minimal changes wrt to the source of your fork, you can simply use patches and/or a custom `CMakeLists.txt`. It is header-only, and vcpkg doesn't want to build tests.

OK

> * The CMake config export is not from the source of your fork, so it should be moved to namespace "unofficial".

What is "CMake config export"? Does it mean `limonp-config.cmake` here? It's auto-generated.

> * I fail to see how the added license and copyright are authorized by the source of your fork.

Yes, `cppjieba` is a port of [[jieba](https://github.com/fxsjy/jieba](https://github.com/fxsjy/jieba), and it is MIT, so I add it.
Because it has no licence, so anyone can be added, right?

> * Please don't close PRs to open new ones for the same topic. Just force-push your updated branch.

OK

**Comment by dg0yt** [comment]
> > * The CMake config export is not from the source of your fork, so it should be moved to namespace "unofficial".
> 
> What is "CMake config export"? Does it mean `limonp-config.cmake` here? It's auto-generated.

It is generated to the letters of your CMakeLists.txt. You have to adjust the instructions:
- Add namespace `unofficial::limonp::`.
- Add prefix `unofficial-` to config file name and installation directory.

> > * I fail to see how the added license and copyright are authorized by the source of your fork.
> 
> Yes, `cppjieba` is a port of [[jieba](https://github.com/fxsjy/jieba](https://github.com/fxsjy/jieba), and it is MIT, so I add it. 

The original repository is https://github.com/yanyiwu/limonp, and there is no indication of license.

> Because it has no licence, so anyone can be added, right?

IANAL. AFAIU it is the contrary, in most legislations: You are not granted any rights. Copyright and everything else remains with the original authors.

File: ports/limonp/portfile.cmake (hunk 3 above)
Comment (FrankXie05) [reply ports/limonp/portfile.cmake:17]: No, it is necessary for the copyright, we can install the readme.md provided upstream.

And set the license field in vcpkg.json to `NULL`

File: ports/limonp/portfile.cmake (hunk 1 above)
Comment (FrankXie05) [reply ports/limonp/portfile.cmake:3]: If there is an official CMakeLists.txt, it must be used. You can add patches to modify the official one.

File: ports/limonp/portfile.cmake (hunk 3 above)
Comment (dg0yt) [reply ports/limonp/portfile.cmake:17]: > And set the license field in vcpkg.json to `NULL`

`null`, with no quotes.

**Comment by FrankXie05** [comment]
As **dg0yt** said, in your [CMakeLists.txt](https://github.com/playgithub/limonp/blob/7f99b0c4c743f3e91e9c912c47a71c23967ef024/CMakeLists.txt#L46) when you export the cmake configuration yourself, the namespace must add `unofficial` flags

File: ports/limonp/portfile.cmake (hunk 3 above)
Comment (playgithub) [reply ports/limonp/portfile.cmake:17]: > No, it is necessary for the copyright, we can install the readme.md provided upstream.

If so, copyright is fake.

**Comment by playgithub** [comment]
I'm going to close the PR now:
* Official libs are better, but the author is not in effective response for PR now.
* [limonp](https://github.com/yanyiwu/limonp) is not intresting by itself, but as a dependy of [cppjieba](https://github.com/yanyiwu/cppjieba), which is a very useful for Chinese text segmentation.

When things are ready, I'd like to port both.

Thanks

**Pull request closed without merging**